k2/util/Logging.cc

```diff
@@ -0,0 +1,57 @@
+// $FILE_PATHNAME $HEADER_FILENAME $HEADER_PATHNAME
+
+// Copyright (c) 2020 Xiaomi Corporation ( authors: Meixu Song )
+//
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/util/Logging.h"
+
+#ifdef K2_USE_GLOG
+// Google glog's api does not have an external function that allows one to check
+// if glog is initialized or not. It does have an internal function - so we are
+// declaring it here. This is a hack but has been used by a bunch of others too.
+namespace google {
+namespace glog_internal_namespace_ {
+bool IsGoogleLoggingInitialized();
+}  // namespace glog_internal_namespace_
+}  // namespace google
+
+namespace k2 {
+bool InitK2Logging(int* argc, char** argv) {
+  if (*argc == 0)
+    return true;
+#if !defined(_MSC_VER)
+  // This trick can only be used on UNIX platforms
+  if (!::google::glog_internal_namespace_::IsGoogleLoggingInitialized())
+#endif
+  {
+    ::google::InitGoogleLogging(argv[0]);
+#if !defined(_MSC_VER)
+    // This is never defined on Windows
+    ::google::InstallFailureSignalHandler();
+#endif
+  }
+  UpdateLoggingLevelsFromFlags();
+  return true;
+}
+
+void UpdateLoggingLevelsFromFlags() {
+  // TODO(meixu): set some default FLAGS/gflags here
+}
+}  // namespace k2
+#else  // !K2_USE_GLOG
+namespace k2 {
+bool InitK2Logging(int *argc, char **argv) {
+  // When doing InitCaffeLogging, we will assume that caffe's flag parser has
```

Comment:
Why does the comment contain `caffe`?
Is the code copied from `caffe`? Please add some comment
in the code to indicate where it is from.

Reply:
the argc,argv stuff may not be necessary as I don't anticipate using k2 much in command line programs.

Reply:
Emm, let me explain this. Before coding, I investigated pytorch, tf, bazel, and other projects that make use of glog, and many alternative logging methods. To find one best practice for logging, some common thought is used here. I try to keep minimal and non-specific to some project codes here. As here, it declares an internal function of glog, which is a common hack as it said, so I leave it as it is. However, this comment shouldn't be here. As the loguru here does parse the arg. The comment is removed now. 
----
To @danpovey :
This function is necessary to init both loggers. At least, to configure the log to stdout or logfile. So I prefer to keep it right now and make the changes we need later.